hotham/src/contexts/render_context.rs

```diff
@@ -71,35 +74,7 @@ pub struct RenderContext {
     pub primitive_map: HashMap<u32, InstancedPrimitive>,
 }
 
-pub struct Shaders {
-    pub vertex_shader: Vec<u32>,
-    pub fragment_shader: Vec<u32>,
-    pub compute_shader: Vec<u32>,
-}
-
-impl Default for Shaders {
-    fn default() -> Self {
-        Self {
-            vertex_shader: VERT.into(),
-            fragment_shader: FRAG.into(),
-            compute_shader: COMPUTE.into(),
-        }
-    }
-}
-
-impl Shaders {
-    pub fn new(
-        vertex_shader: Vec<u32>,
-        fragment_shader: Vec<u32>,
-        compute_shader: Vec<u32>,
-    ) -> Self {
-        Self {
-            vertex_shader,
-            fragment_shader,
-            compute_shader,
-        }
-    }
-}
+pub type Shaders = HashMap<String, Vec<u32>>;
```

Comment:
I think we lose something by getting rid of the struct here! `RenderContext` has *only* three shaders that all do very specific things - having an arbitrary number of them, identified by a string doesn't seem to meet that goal.